python/sdist/amici/petab/parameter_mapping.py

```diff
@@ -352,7 +352,11 @@ def create_parameter_mapping(
     if petab_problem.model.type_id == MODEL_TYPE_SBML:
         import libsbml
 
-        if petab_problem.sbml_document:
+        # v1 guard
+        if (
+            isinstance(petab_problem, petab.Problem)
+            and petab_problem.sbml_document
+        ):
```

Comment:
```suggestion
        if petab_problem.model.sbml_document:
```
Works for v1 and v2.